### PR TITLE
Added solution to leetcode 134

### LINKED ISSUE #2627: Leetcode solution 134 is Missing
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

image

### Use Case

Leetcode solution 134 is Missing

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by Utsavladia**
## Related Issue
Fix #2627 

[Cite any related issue(s) this pull request addresses. If none, simply state "None”]

## Description
Added solution to leetcode 134
[Please include a brief description of the changes or features added]

## Type of PR

- [x] Bug fix
- [ ] Feature enhancement
- [ ] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)
image

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [ ] I have performed a self-review of my code.
- [ ] I have read and followed the Contribution Guidelines.
- [ ] I have tested the changes thoroughly before submitting this pull request.
- [ ] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have followed the code style guidelines of this project.
- [ ] I have checked for any existing open issues that my pull request may address.
- [ ] I have ensured that my changes do not break any existing functionality.
- [ ] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [ ] I have read the resources for guidance listed below.
- [ ] I have followed security best practices in my code changes.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [ ] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [ ] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [ ] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [ ] [Markdown Guide](https://www.markdownguide.org/)
- [ ] [MDX Documentation](https://mdxjs.com/docs/)
- [ ] [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- [ ] [Tabs: Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [ ] [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [ ] [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex) 

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

File: dsa-solutions/lc-solutions/0100-0199/0134-gas-station.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  - DP
+  - BackTracking
+description: "This is a solution to the Gas station problem on LeetCode."
+---
+
+## Problem Description
+
+There are n gas stations along a circular route, where the amount of gas at the ith station is gas[i].
+
+You have a car with an unlimited gas tank and it costs cost[i] of gas to travel from the ith station to its next (i + 1)th station. You begin the journey with an empty tank at one of the gas stations.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0134-gas-station.md:16]: replace `(i + 1)th` to `$(i + 1)^{th}$` for best practices

File: dsa-solutions/lc-solutions/0100-0199/0134-gas-station.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+You can't start at station 0 or 1, as there is not enough gas to travel to the next station.
+Let's start at station 2 and fill up with 4 unit of gas. Your tank = 0 + 4 = 4
+Travel to station 0. Your tank = 4 - 3 + 2 = 3
+Travel to station 1. Your tank = 3 - 3 + 3 = 3
+You cannot travel back to station 2, as it requires 4 unit of gas but you only have 3.
+Therefore, you can't travel around the circuit once no matter where you start.
+```
+
+### Constraints:
+
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0134-gas-station.md:51]: replace 

```md

- `n == gas.length == cost.length`
- `1 <= n <= 105`
- `0 <= gas[i], cost[i] <= 104`
```
to

```md

- $n == \text{gas.length} == \text{cost.length}$
- $1 \leq \text{n} \leq 105$
- $0 \leq \text{gas[i], cost[i]} \leq 104$
```

File: dsa-solutions/lc-solutions/0100-0199/0134-gas-station.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            for j in range(start+1, len(cost)):
+                if cost[i] > gas[i]:
+                    print("inside", j)
+        return -1
+```
+
+
+### Complexity
+
+- **Time Complexity:** O(n) - A single pass through the gas stations.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0134-gas-station.md:153]: replace `O(n)` to `$O(n)$`

File: dsa-solutions/lc-solutions/0100-0199/0134-gas-station.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+- **Space Complexity:** O(1) - Constant space usage.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0134-gas-station.md:154]: replace `O(1)` to `$O(1)$`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Comment by Utsavladia** [comment]
@Ajay-Dhangar  sure I'll do the sugggested changes...

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**